gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/InitialImageOperation.java

```diff
@@ -232,6 +232,16 @@ private GIIStatus reportGIIStatus() {
     }
   }
 
+  private boolean hasLockedBucket = false;
+
+  private void unlockTargetBucketLock(InternalDistributedMember recipient) {
+    if (region instanceof BucketRegion && hasLockedBucket) {
```

Comment:
In fact, we can remove the check as hasLockedBucket will only be true for BucketRegions. This check is redundant.